### ISSUE TITLE
Make public ViewModelLocationProvider.GetViewModelForView method

### ISSUE THREAD

**Issue by YehudahA** [PrismLibrary/Prism#347]
I want to use the same logic of ViewModelLocationProvider for other services [DialogService and more],
Please, if you can make it public, it's can be very helpful.

I suggest to expose something like this:
``` csharp
public static object GetViewModelForView(Type viewType)
```

**Comment by brianlagunas** [comment]
I'm not sure I follow.  A DialogService won't have a ViewModel.  The view it shows will, but not the service.  Can you explain your scenario in more detail please?

**Comment by YehudahA** [comment]
You're right! I was confused! In case of dialogService I need the opposite method: GetViewForViewModel.
But I still need GetViewModelForView for other services, and make it public can be very helpful.

**Comment by brianlagunas** [comment]
How would you use it?  The method only creates ViewModels for Views.  It doesn't cache them, so if you are calling this method, you get a different instance.  Can you explain how this would be helpful in your scenario?

**Comment by YehudahA** [comment]
I'm writing a navigation service that load the viewModels async:

``` csharp
// TODO: GetViewModelForView
var viewModel = await Task.Run( ()=> GetViewModelForView(newViewType);
newView.DataContext = viewModel;
```

**Comment by brianlagunas** [comment]
Well, actually after looking at the method GetViewModelForView, it turns out it doesn't ever create ViewModels.

https://github.com/PrismLibrary/Prism/blob/master/Source/Prism/Mvvm/ViewModelLocationProvider.cs#L107

So this methods doesn't help you out.

**Comment by Haukinger** [comment]
I cloned the ViewModelLocationProvider and added a public GetViewModelType to be able to resolve the viewmodel in INavigationAware.OnNavigatedTo using parameters from the NavigationContext injected into the viewmodel's constructor.

It looks a bit weird, but I couldn't come up with a better solution. It would be great, though, if we could make the ViewModelLocationProvider itself optionally pass a NavigationContext to its viewmodel factory somehow.

**Comment by brianlagunas** [comment]
The VMLP is cross platform and has a single responsibility.  Not only should it never know about a NavigationContext, bit the other platforms do not have such an object.

I am really confused on why you are trying to use the VMLP to resolve ViewModels from with your ViewModel.  Just use the container, that's its job.

**Comment by Haukinger** [comment]
I shortened too much, perhaps. I've got a bunch of view types, each of which has it's own viewmodel type. I can navigate to the views, and the vmlp creates the viewmodels by using the container. This works fine until I need to pass information from the navigation context to the constructor of the viewmodel.
So I built my own "vmlp" into a base class for all these views. It does pretty much the same thing as the regular vmlp, but defers the viewmodel creation until a OnNavigateTo is called on the view base class. That works fine, too, but I'm not happy with the base class approach, I'd rather prefer to attach something like the vmlp...
